src/serialmonitor/serialportctrl.ts

```diff
@@ -75,47 +48,51 @@ export class SerialPortCtrl {
     this._currentTimestampFormat = timestampFormat;
   }
 
-  /*
-  * Return true if child proccess is currently running
-  */
   public get isActive(): boolean {
-    return this._child ? true : false;
+    return this._port?.isOpen ?? false;
   }
 
   public get currentPort(): string {
     return this._currentPort;
   }
 
-  public open(): Promise<void> {
+  public async open(): Promise<void> {
     this._bufferedOutputChannel.appendLine(`[Starting] Opening the serial port - ${this._currentPort}`);
     this.showOutputChannel();
 
-    if (this._child) {
-        this.stop();
-    }
-
-    return new Promise((resolve, reject) => {
-        this._child = spawn(SerialPortCtrl._serialCliPath,
-            ["open", this._currentPort, "-b", this._currentBaudRate.toString(), "-t", this._currentTimestampFormat, "--json"])
+    if (this.isActive) { await this.close(); }
 
-        this._child.on("error", (err) => {
-            reject(err)
+    await new Promise<void>((resolve, reject) => {
+      this._port = new SerialPort(
+        { autoOpen: false, path: this._currentPort, baudRate: this._currentBaudRate },
+        (err) => {
+          if (err) { reject(err); }
         });
+      this._port.open((err) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve();
+        }
+      });
+    });
 
-        this._child.stdout.on("data", (data) => {
-            if (this.isActive) {
-                const jsonObj = JSON.parse(data.toString())
-                this._bufferedOutputChannel.append(jsonObj["timestamp"] + jsonObj["payload"]);
-            }
-        });
-        // TODO: add message check to ensure _child spawned without errors
-        resolve();
-        // The spawn event is only supported in node v15+ vscode
-        // this._child.on("spawn", (spawn) => {
-        //     resolve();
-        // });
-
-     });
+    let lastDataEndedWithNewline = true;
+    this._port.on("data", (data) => {
```

Comment:
This likely answers my above question, just an event listener on the port, so we react on any data.

Reply:
Yep, this is actually just matching the Node `Stream` interface, so `data` will fire whenever data is available from the stream.